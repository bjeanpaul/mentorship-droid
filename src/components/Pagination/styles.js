import { StyleSheet } from 'react-native';
import colors from 'src/constants/colors';
import { FONT, FONT_WEIGHT } from 'src/constants/styles';

export default StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5,
  },
  buttonText: {
    fontFamily: FONT.BOLD,
    fontWeight: FONT_WEIGHT.BOLD,
    fontSize: 13,
    letterSpacing: 1,
    marginTop: 15,
    marginBottom: 17,
  },
  labelDefault: {
    color: colors.BUTTON_LABEL,
  },
  labelWhite: {
    color: colors.BUTTON_ALT_LABEL,
  },
  labelDisabled: {
    color: colors.BUTTON_DISABLED_LABEL,
  },
});
