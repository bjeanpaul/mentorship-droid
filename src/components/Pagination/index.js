import { noop, debounce } from 'lodash';
import React, { PropTypes, Component } from 'react';
import { View } from 'react-native';

import Button from 'src/components/Button';
import Icon from 'src/components/Icon';
import styles from './styles';

import { PAGING_DEBOUNCE_INTERVAL } from 'src/constants/behavior';


const debouncePager = fn => debounce(fn, PAGING_DEBOUNCE_INTERVAL, {
  leading: true,
  trailing: false,
});


class Pagination extends Component {
  static defaultProps = {
    onBackPress: noop,
    onNextPress: noop,
    onDonePress: noop,
    last: false,
  }

  static propTypes = {
    onBackPress: PropTypes.func,
    onNextPress: PropTypes.func,
    onDonePress: PropTypes.func,
    backDisabled: PropTypes.bool,
    disabled: PropTypes.bool,
    last: PropTypes.bool,
  }

  constructor(...args) {
    super(...args);
    this.onDonePress = debouncePager(this.onDonePress.bind(this));
    this.onNextPress = debouncePager(this.onNextPress.bind(this));
    this.onBackPress = debouncePager(this.onBackPress.bind(this));
  }

  onDonePress() {
    this.props.onDonePress();
  }

  onNextPress() {
    this.props.onNextPress();
  }

  onBackPress() {
    this.props.onBackPress();
  }

  render() {
    const {
      backDisabled,
      disabled,
      last,
    } = this.props;

    return (
      <View style={styles.container}>
        <Button
          uid="back"
          theme={Button.themes.transparent}
          layout={[Button.layouts.inline, styles.backButtonLayout]}
          onPress={this.onBackPress}
          disabled={backDisabled}
        >
          <Icon type={Icon.types.backOrange} />
        </Button>
        {
        !last
        ? <Button
          uid="next"
          theme={Button.themes.transparent}
          layout={[Button.layouts.inline, styles.nextButtonLayout]}
          onPress={this.onNextPress}
          disabled={disabled}
        >
          NEXT
        </Button>
        : <Button
          uid="done"
          theme={Button.themes.transparent}
          layout={[Button.layouts.inline, styles.nextButtonLayout]}
          onPress={this.onDonePress}
          disabled={disabled}
        >
          DONE
        </Button>
        }
      </View>
    );
  }
}


export default Pagination;
