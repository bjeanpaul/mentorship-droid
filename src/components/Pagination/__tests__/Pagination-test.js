import React from 'react';
import Pagination from 'src/components/Pagination';


describe('Button', () => {
  function foo() {}
  function bar() {}

  it('should render props correctly', () => {
    expect(render(
      <Pagination
        onBackPress={foo}
        onNextPress={bar}
      />
    )).toMatchSnapshot();
  });

  it('should be disabled', () => {
    expect(render(
      <Pagination
        onBackPress={foo}
        onNextPress={bar}
        disabled
      />
    )).toMatchSnapshot();
  });

  it('should be able to tap and fire `onNextPress`', () => {
    const mockFn = jest.fn();
    const el = shallow(
      <Pagination
        onBackPress={foo}
        onNextPress={mockFn}
      />
    );
    el.find('Button').at(1).simulate('press');
    expect(mockFn).toBeCalled();
  });

  it('should be able to tap and fire `onBackPress`', () => {
    const mockFn = jest.fn();
    const el = shallow(
      <Pagination
        onBackPress={mockFn}
        onNextPress={bar}
      />
    );
    el.find('Button').at(0).simulate('press');
    expect(mockFn).toBeCalled();
  });
});
