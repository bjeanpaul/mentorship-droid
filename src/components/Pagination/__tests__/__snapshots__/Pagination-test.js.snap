exports[`Pagination should disable its done button if disabled is true 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "margin": 16,
    }
  }>
  <TouchableNativeFeedback
    onPress={[Function]}>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "paddingBottom": 17,
            "paddingTop": 15,
          },
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#f77040",
            "borderWidth": 1,
          },
          Array [
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            Object {
              "marginRight": 16,
              "width": 58,
            },
          ],
          undefined,
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "fontSize": 13,
              "textAlign": "center",
            },
            Array [
              Object {
                "fontFamily": "Brandon Text_bold",
                "fontSize": 13,
                "fontWeight": "bold",
                "letterSpacing": 1,
              },
              Object {
                "color": "#f77040",
              },
              undefined,
              undefined,
            ],
          ]
        }>
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 24,
                "width": 24,
              },
              Object {},
            ]
          } />
      </Text>
    </View>
  </TouchableNativeFeedback>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
          "marginHorizontal": 16,
          "paddingBottom": 17,
          "paddingTop": 15,
        },
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#f77040",
          "borderWidth": 1,
        },
        Array [
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          Object {
            "flex": 1,
          },
        ],
        undefined,
        Array [
          Object {
            "borderColor": "#dfe5e6",
          },
          undefined,
        ],
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
          Object {
            "fontSize": 13,
            "textAlign": "center",
          },
          Array [
            Object {
              "fontFamily": "Brandon Text_bold",
              "fontSize": 13,
              "fontWeight": "bold",
              "letterSpacing": 1,
            },
            Object {
              "color": "#f77040",
            },
            undefined,
            Array [
              Object {
                "color": "#dfe5e6",
              },
              undefined,
            ],
          ],
        ]
      }>
      DONE
    </Text>
  </View>
</View>
`;

exports[`Pagination should disable its next button if disabled is true 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "margin": 16,
    }
  }>
  <TouchableNativeFeedback
    onPress={[Function]}>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "paddingBottom": 17,
            "paddingTop": 15,
          },
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#f77040",
            "borderWidth": 1,
          },
          Array [
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            Object {
              "marginRight": 16,
              "width": 58,
            },
          ],
          undefined,
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "fontSize": 13,
              "textAlign": "center",
            },
            Array [
              Object {
                "fontFamily": "Brandon Text_bold",
                "fontSize": 13,
                "fontWeight": "bold",
                "letterSpacing": 1,
              },
              Object {
                "color": "#f77040",
              },
              undefined,
              undefined,
            ],
          ]
        }>
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 24,
                "width": 24,
              },
              Object {},
            ]
          } />
      </Text>
    </View>
  </TouchableNativeFeedback>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
          "marginHorizontal": 16,
          "paddingBottom": 17,
          "paddingTop": 15,
        },
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#f77040",
          "borderWidth": 1,
        },
        Array [
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          Object {
            "flex": 1,
          },
        ],
        undefined,
        Array [
          Object {
            "borderColor": "#dfe5e6",
          },
          undefined,
        ],
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
          Object {
            "fontSize": 13,
            "textAlign": "center",
          },
          Array [
            Object {
              "fontFamily": "Brandon Text_bold",
              "fontSize": 13,
              "fontWeight": "bold",
              "letterSpacing": 1,
            },
            Object {
              "color": "#f77040",
            },
            undefined,
            Array [
              Object {
                "color": "#dfe5e6",
              },
              undefined,
            ],
          ],
        ]
      }>
      NEXT
    </Text>
  </View>
</View>
`;

exports[`Pagination should render a done button if last is true 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "margin": 16,
    }
  }>
  <TouchableNativeFeedback
    onPress={[Function]}>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "paddingBottom": 17,
            "paddingTop": 15,
          },
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#f77040",
            "borderWidth": 1,
          },
          Array [
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            Object {
              "marginRight": 16,
              "width": 58,
            },
          ],
          undefined,
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "fontSize": 13,
              "textAlign": "center",
            },
            Array [
              Object {
                "fontFamily": "Brandon Text_bold",
                "fontSize": 13,
                "fontWeight": "bold",
                "letterSpacing": 1,
              },
              Object {
                "color": "#f77040",
              },
              undefined,
              undefined,
            ],
          ]
        }>
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 24,
                "width": 24,
              },
              Object {},
            ]
          } />
      </Text>
    </View>
  </TouchableNativeFeedback>
  <TouchableNativeFeedback
    onPress={[Function]}>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "paddingBottom": 17,
            "paddingTop": 15,
          },
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#f77040",
            "borderWidth": 1,
          },
          Array [
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            Object {
              "flex": 1,
            },
          ],
          undefined,
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "fontSize": 13,
              "textAlign": "center",
            },
            Array [
              Object {
                "fontFamily": "Brandon Text_bold",
                "fontSize": 13,
                "fontWeight": "bold",
                "letterSpacing": 1,
              },
              Object {
                "color": "#f77040",
              },
              undefined,
              undefined,
            ],
          ]
        }>
        DONE
      </Text>
    </View>
  </TouchableNativeFeedback>
</View>
`;

exports[`Pagination should render props correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "margin": 16,
    }
  }>
  <TouchableNativeFeedback
    onPress={[Function]}>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "paddingBottom": 17,
            "paddingTop": 15,
          },
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#f77040",
            "borderWidth": 1,
          },
          Array [
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            Object {
              "marginRight": 16,
              "width": 58,
            },
          ],
          undefined,
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "fontSize": 13,
              "textAlign": "center",
            },
            Array [
              Object {
                "fontFamily": "Brandon Text_bold",
                "fontSize": 13,
                "fontWeight": "bold",
                "letterSpacing": 1,
              },
              Object {
                "color": "#f77040",
              },
              undefined,
              undefined,
            ],
          ]
        }>
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 24,
                "width": 24,
              },
              Object {},
            ]
          } />
      </Text>
    </View>
  </TouchableNativeFeedback>
  <TouchableNativeFeedback
    onPress={[Function]}>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "paddingBottom": 17,
            "paddingTop": 15,
          },
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#f77040",
            "borderWidth": 1,
          },
          Array [
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            Object {
              "flex": 1,
            },
          ],
          undefined,
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "fontSize": 13,
              "textAlign": "center",
            },
            Array [
              Object {
                "fontFamily": "Brandon Text_bold",
                "fontSize": 13,
                "fontWeight": "bold",
                "letterSpacing": 1,
              },
              Object {
                "color": "#f77040",
              },
              undefined,
              undefined,
            ],
          ]
        }>
        NEXT
      </Text>
    </View>
  </TouchableNativeFeedback>
</View>
`;

exports[`Pagination should support disabling its back button 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "margin": 16,
    }
  }>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "justifyContent": "center",
          "marginHorizontal": 16,
          "paddingBottom": 17,
          "paddingTop": 15,
        },
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#f77040",
          "borderWidth": 1,
        },
        Array [
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          },
          Object {
            "marginRight": 16,
            "width": 58,
          },
        ],
        undefined,
        Array [
          Object {
            "borderColor": "#dfe5e6",
          },
          undefined,
        ],
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
          Object {
            "fontSize": 13,
            "textAlign": "center",
          },
          Array [
            Object {
              "fontFamily": "Brandon Text_bold",
              "fontSize": 13,
              "fontWeight": "bold",
              "letterSpacing": 1,
            },
            Object {
              "color": "#f77040",
            },
            undefined,
            Array [
              Object {
                "color": "#dfe5e6",
              },
              undefined,
            ],
          ],
        ]
      }>
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": 24,
              "width": 24,
            },
            Object {},
          ]
        } />
    </Text>
  </View>
  <TouchableNativeFeedback
    onPress={[Function]}>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "justifyContent": "center",
            "marginHorizontal": 16,
            "paddingBottom": 17,
            "paddingTop": 15,
          },
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#f77040",
            "borderWidth": 1,
          },
          Array [
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            Object {
              "flex": 1,
            },
          ],
          undefined,
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "fontSize": 13,
              "textAlign": "center",
            },
            Array [
              Object {
                "fontFamily": "Brandon Text_bold",
                "fontSize": 13,
                "fontWeight": "bold",
                "letterSpacing": 1,
              },
              Object {
                "color": "#f77040",
              },
              undefined,
              undefined,
            ],
          ]
        }>
        NEXT
      </Text>
    </View>
  </TouchableNativeFeedback>
</View>
`;
